### PR TITLE
{ACR} Change: az acr login support for resource group and skip resource validation for non acr source registry in az acr import

### PR DESCRIPTION
**Pull request by siby-george**
**Related command**
`az acr login` and `az acr import`

**Description**
`az acr login`
added optional resource group argument to skip subscription level resource scan which fans out to all region and can cause failure if any of arm region have outage.

`az acr import`
added a check to avoid subscription scan if source registry is not an ACR registry. no observable changes for customer.

**Testing Guide** 
az acr login -n registry name -g resoucegroup
az acr login -n registryname
az acr import -n myregistry --source sourceregistry.azurecr.io/sourcerepository:sourcetag
az acr import -n myregistry --source docker.io/library/hello-world:latest

**History Notes**
[ACR] `az acr login`: Support optional resource group argument to skip subscription resource scan

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/acr/import.py [hunk 1]
```diff
@@ -77,7 +77,8 @@ def acr_import(cmd,  # pylint: disable=too-many-locals
                                   credentials=ImportSourceCredentials(password=source_registry_password,
                                                                       username=source_registry_username))
         else:
-            registry = get_registry_from_name_or_login_server(cmd.cli_ctx, registry_uri)
+            if registry_uri.endswith('azurecr.io') or registry_uri.endswith('.azurecr-test.io'):
```
Comment (rosanch) [inline src/azure-cli/azure/cli/command_modules/acr/import.py:80]: What about National Clouds?

**Review by rosanch — CHANGES_REQUESTED** [review]
LGTM just plan for NC

File: src/azure-cli/azure/cli/command_modules/acr/import.py (hunk 1 above)
Comment (siby-george) [reply src/azure-cli/azure/cli/command_modules/acr/import.py]: Fixed by getting login_server_suffix from cli context

**Review by rosanch — APPROVED** [review]
LGTM

**Comment by yanzhudd** [comment]
please fix the style check

**Review by rosanch — APPROVED** [review]

**Comment by yanzhudd** [comment]
please add test for this change to verify if the change meets your expectation.

**Comment by siby-george** [comment]
> please add test for this change to verify if the change meets your expectation.

existing test case already cover these path

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**